Add some documentation about the webhook (#2158)
elastic/cloud-on-k8s: docs/troubleshooting.asciidoc

On startup, the operator deploys an https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/[admission webhook] that points to the operator's service. If this is inaccessible, you may see errors in your Kubernetes API server logs indicating that it cannot reach the service. A common cause may be that the operator pods are failing to start for some reason, or that the control plane is isolated from the operator pod by some mechanism (for instance via network policies or running the control plane externally as in https://github.com/elastic/cloud-on-k8s/issues/896#issuecomment-507224945[issue #869] and https://github.com/elastic/cloud-on-k8s/issues/1369[issue #1369]).

> If you think that the validation is not done correctly you can also change the `failurePolicy` of the webhook configuration to `Fail`, which will cause creations to error out if there is an error contacting the webhook.

Reviewer: ```suggestion
You can also change the https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#failure-policy[`failurePolicy`] of the webhook configuration to `Fail`, which will cause creations and updates to error out if there is an error contacting the webhook.
```

I don't know if the monospace font works in a href though. But I do think that the first part of the sentence can be removed for clarity.

Author: I'll commit your suggestion and check the result on the CI

Author: > I don't know if the monospace font works in a href though

Looks like it is working as expected:
![image](https://user-images.githubusercontent.com/976373/70655019-c4efbe00-1c57-11ea-8394-016c6d6ff317.png)

Reviewer: Sweet, good to know
